### PR TITLE
Added Campaign Subscription functionality, and tests for them.

### PR DESCRIPTION
**Pull request by ekoval**
Added Campaign Subscription functionality, and tests for it.
https://github.com/yola/production/issues/2116

### CODE REVIEW

**Comment by snitch** [comment]
There are 16 errors:

* sitewit/services.py, line 174 - E125 continuation line with same indent as next logical line
* sitewit/services.py, line 211 - E127 continuation line over-indented for visual indent
* sitewit/services.py, line 223 - E127 continuation line over-indented for visual indent
* sitewit/services.py, line 226 - E125 continuation line with same indent as next logical line
* sitewit/services.py, line 250 - E125 continuation line with same indent as next logical line
* sitewit/services.py, line 274 - E125 continuation line with same indent as next logical line
* sitewit/services.py, line 297 - E125 continuation line with same indent as next logical line
* sitewit/services.py, line 316 - E501 line too long (80 > 79 characters)
* tests/campaigns/base.py, line 13 - E126 continuation line over-indented for hanging indent
* tests/campaigns/test_integration.py, line 79 - E501 line too long (82 > 79 characters)
* tests/campaigns/test_integration.py, line 290 - F841 local variable 'campaign' is assigned to but never used
* tests/campaigns/test_integration.py, line 294 - E226 missing whitespace around arithmetic operator
* tests/campaigns/test_integration.py, line 299 - E226 missing whitespace around arithmetic operator
* tests/campaigns/test_integration.py, line 333 - F841 local variable 'campaign' is assigned to but never used
* tests/campaigns/test_integration.py, line 337 - E226 missing whitespace around arithmetic operator
* tests/campaigns/test_integration.py, line 342 - E226 missing whitespace around arithmetic operator

**Comment by snitch** [comment]
There are 16 errors:

* sitewit/services.py, line 174 - E125 continuation line with same indent as next logical line
* sitewit/services.py, line 211 - E127 continuation line over-indented for visual indent
* sitewit/services.py, line 223 - E127 continuation line over-indented for visual indent
* sitewit/services.py, line 226 - E125 continuation line with same indent as next logical line
* sitewit/services.py, line 250 - E125 continuation line with same indent as next logical line
* sitewit/services.py, line 274 - E125 continuation line with same indent as next logical line
* sitewit/services.py, line 297 - E125 continuation line with same indent as next logical line
* sitewit/services.py, line 316 - E501 line too long (80 > 79 characters)
* tests/campaigns/base.py, line 13 - E126 continuation line over-indented for hanging indent
* tests/campaigns/test_integration.py, line 79 - E501 line too long (82 > 79 characters)
* tests/campaigns/test_integration.py, line 290 - F841 local variable 'campaign' is assigned to but never used
* tests/campaigns/test_integration.py, line 294 - E226 missing whitespace around arithmetic operator
* tests/campaigns/test_integration.py, line 299 - E226 missing whitespace around arithmetic operator
* tests/campaigns/test_integration.py, line 333 - F841 local variable 'campaign' is assigned to but never used
* tests/campaigns/test_integration.py, line 337 - E226 missing whitespace around arithmetic operator
* tests/campaigns/test_integration.py, line 342 - E226 missing whitespace around arithmetic operator

File: tests/campaigns/test_integration.py [hunk 1]
```diff
@@ -57,10 +61,301 @@ class TestDeleteCampaign(CampaignTestCase):
...
+
+        self.budget = subscription['subscription']['budget']
+        currency = subscription['subscription']['currency']
+
+        if subscription['status'] != 'Active':
+            service.renew_campaign_subscription(
+                self.account_token, self.campaign_id, self.budget, currency)
+
+        self.result = service.upgrade_campaign_subscription(
+            self.account_token, self.campaign_id, self.budget+10, currency)
```
Comment (snitch) [inline tests/campaigns/test_integration.py]: E226 missing whitespace around arithmetic operator

File: tests/campaigns/test_integration.py [hunk 2]
```diff
@@ -57,10 +61,301 @@ class TestDeleteCampaign(CampaignTestCase):
...
+
+    def test_upgraded_campaign_is_returned(self):
+        self.assertEqual(self.result['id'], self.campaign_id)
+        self.assertEqual(
+            self.result['subscription']['budget'], self.budget+10)
```
Comment (snitch) [inline tests/campaigns/test_integration.py]: E226 missing whitespace around arithmetic operator

File: tests/campaigns/test_integration.py [hunk 3]
```diff
@@ -57,10 +61,301 @@ class TestDeleteCampaign(CampaignTestCase):
...
+
+        self.budget = subscription['subscription']['budget']
+        currency = subscription['subscription']['currency']
+
+        if subscription['status'] != 'Active':
+            service.renew_campaign_subscription(
+                self.account_token, self.campaign_id, self.budget, currency)
+
+        self.result = service.downgrade_campaign_subscription(
+            self.account_token, self.campaign_id, self.budget-10, currency)
```
Comment (snitch) [inline tests/campaigns/test_integration.py]: E226 missing whitespace around arithmetic operator

File: tests/campaigns/test_integration.py [hunk 4]
```diff
@@ -57,10 +61,301 @@ class TestDeleteCampaign(CampaignTestCase):
...
+
+    def test_downgraded_campaign_is_returned(self):
+        self.assertEqual(self.result['id'], self.campaign_id)
+        self.assertEqual(
+            self.result['subscription']['budget'], self.budget-10)
```
Comment (snitch) [inline tests/campaigns/test_integration.py]: E226 missing whitespace around arithmetic operator

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: sitewit/services.py [hunk 5]
```diff
@@ -167,5 +167,156 @@ def delete_campaign(self, account_token, campaign_id):
...
+            Please see response format here:
+            https://sandboxpapi.sitewit.com/Help/Api/
+            GET-api-subscription-campaign-id
+        """
+        data = {'campaignId': campaign_id,
+                'budget': new_budget,
+                'currency': currency}
+
+        return self.put(
+            '/api/subscription/campaign/downgrade', data,
```
Comment (stefanor) [inline sitewit/services.py]: Upgrade has a trailing slash

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: sitewit/services.py [hunk 6]
```diff
@@ -167,5 +167,156 @@ def delete_campaign(self, account_token, campaign_id):
...
+        """
+        data = {'campaignId': campaign_id,
+                'budget': new_budget,
+                'currency': currency}
+
+        return self.put(
+            '/api/subscription/campaign/downgrade/', data,
+            headers=self._get_auth_header(account_token)).json()
+
+    def renew_campaign_subscription(self, account_token, campaign_id,
```
Comment (RayeN) [inline sitewit/services.py]: `resume_campaign_subscription` IMO sounds better.

File: sitewit/services.py [hunk 7]
```diff
@@ -167,5 +167,156 @@ def delete_campaign(self, account_token, campaign_id):
...
+        data = {'campaignId': campaign_id,
+                'budget': new_budget,
+                'currency': currency}
+
+        return self.put(
+            '/api/subscription/campaign/downgrade/', data,
+            headers=self._get_auth_header(account_token)).json()
+
+    def resume_campaign_subscription(self, account_token, campaign_id,
+                                    new_budget, currency):
```
Comment (snitch) [inline sitewit/services.py]: E128 continuation line under-indented for visual indent

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by RayeN** [comment]
:+1: Looks sane

**Pull request merged**